File: js/web/examples/monitor.mjs
import {
  ClientBuilder,
  LogLevel,
  Monitor,
  RelayUrl,
  initLogger,
} from "../dist/index.js";

class MonitorHandler {
  async relayStatusChanged(relayUrl, status) {
    console.log(`Relay ${relayUrl} status changed to ${status}`);
  }
}

export async function monitorExample() {
  initLogger(LogLevel.Debug);
  const monitor = new Monitor();
  const client = new ClientBuilder().monitor(monitor).build();
  for (const url of [
    "wss://relay.damus.io",
    "wss://nostr.mom",
    "wss://nostr.oxtr.dev",
  ]) {
    await client.addRelay(RelayUrl.parse(url));
  }
  await client.connect();
  await monitor.handleNotifications(new MonitorHandler());
}

File: kmp/nostr-sdk-kmp/examples/monitor.kt
package org.nostrdevkit.examples

import org.nostrdevkit.sdk.*

private class MonitorHandler : HandleMonitorNotification {
    override suspend fun relayStatusChanged(relayUrl: RelayUrl, status: RelayStatus) {
        println("Relay $relayUrl status changed to $status")
    }
}

suspend fun monitorExample() {
    initLogger(LogLevel.DEBUG)

    val monitor = Monitor()
    val client = ClientBuilder().monitor(monitor).build()
    for (url in listOf("wss://relay.damus.io", "wss://nostr.mom", "wss://nostr.oxtr.dev")) {
        client.addRelay(RelayUrl.parse(url))
    }
    client.connect()
    monitor.handleNotifications(MonitorHandler())
}
